code/modules/mob/emote.dm

```diff
@@ -1,132 +1,89 @@
-#define EMOTE_COOLDOWN 20		//Time in deciseconds that the cooldown lasts
-
-//Emote Cooldown System (it's so simple!)
-/mob/proc/handle_emote_CD(cooldown = EMOTE_COOLDOWN)
-	if(emote_cd == 2) return 1			// Cooldown emotes were disabled by an admin, prevent use
-	if(src.emote_cd == 1) return 1		// Already on CD, prevent use
-
-	src.emote_cd = 1		// Starting cooldown
-	spawn(cooldown)
-		if(emote_cd == 2) return 1		// Don't reset if cooldown emotes were disabled by an admin during the cooldown
-		src.emote_cd = 0				// Cooldown complete, ready for more!
-
-	return 0		// Proceed with emote
-//--FalseIncarnate
-
-/mob/proc/handle_emote_param(var/target, var/not_self, var/vicinity, var/return_mob) //Only returns not null if the target param is valid.
-	var/view_vicinity = vicinity ? vicinity : null									 //not_self means we'll only return if target is valid and not us
-	if(target)																		 //vicinity is the distance passed to the view proc.
-		for(var/mob/A in view(view_vicinity, null))									 //if set, return_mob will cause this proc to return the mob instead of just its name if the target is valid.
-			if(target == A.name && (!not_self || (not_self && target != name)))
-				if(return_mob)
-					return A
-				else
-					return target
-
-// All mobs should have custom emote, really..
-/mob/proc/custom_emote(var/m_type=1,var/message = null)
-
-	if(stat || !use_me && usr == src)
-		if(usr)
-			to_chat(usr, "You are unable to emote.")
+/mob/proc/emote(var/act, var/m_type, var/message)
+	act = lowertext(act)
+	var/param = message
+	var/custom_param = findchar(act, "-")
+	if(custom_param)
+		param = copytext(act, custom_param + 1, length(act) + 1)
+		act = copytext(act, 1, custom_param)
+
+	var/datum/emote/E = emote_list[act]
+	if(!E)
+		to_chat(src, "<span class='notice'>Unusable emote '[act]'. Say *help for a list.</span>")
 		return
-
-	var/muzzled = istype(src.wear_mask, /obj/item/clothing/mask/muzzle)
-	if(m_type == 2 && muzzled) return
-
-	var/input
-	if(!message)
-		input = sanitize(copytext(input(src,"Choose an emote to display.") as text|null,1,MAX_MESSAGE_LEN))
+	E.run_emote(src, param, m_type)
+	
+
+/mob/proc/custom_emote(var/m_type=EMOTE_VISIBLE, var/message=null)
+	emote("me", m_type, message)
+
+/datum/emote/mob/help
+	key = "help"
+	stat_allowed = DEAD
+
+/datum/emote/mob/help/run_emote(mob/user, params)
+	var/list/keys = list()
+	var/prefix = "Available emotes (you can use them with say \"*emote\"): "
+
+	for(var/k in user.emote_list)
+		var/datum/emote/E = user.emote_list[k]
+		if(E.key in keys)
+			continue
+		if(E.can_run_emote(user, status_check = FALSE))
+			keys += E.key
+
+	keys = sortList(keys)
+
+	message = "[prefix] [jointext(keys, ", ")]."
+
+	to_chat(user, message)
+	return TRUE
+
+/datum/emote/mob/custom
+	key = "me"
+	key_third_person = "custom"
+	punct = ""
+	message = null
+
+/datum/emote/mob/custom/proc/check_invalid(mob/user, input)
+	. = TRUE
+	if(copytext(input,1,5) == "says")
```

Comment:
So much snowflake nope right here. This isn't something that the emote should be trying to prevent.